src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -387,28 +388,85 @@ public static string ConcatAsHexSuffix(string str, char separator, uint number)
                 length += str.Length;
             }
 
-            return string.Create(length, (str, separator, number), (buffer, tuple) =>
+            return string.Create(length, (str, separator, number), s_populateSpanWithHexSuffix);
+        }
+
+        private static void PopulateSpanWithHexSuffix(Span<char> buffer, (string str, char separator, uint number) tuple)
+        {
+            var (tupleStr, tupleSeparator, tupleNumber) = tuple;
+
+            var i = 0;
+            if (tupleStr != null)
             {
-                var (tupleStr, tupleSeparator, tupleNumber) = tuple;
-                char[] encode16Chars = s_encode16Chars;
+                tupleStr.AsSpan().CopyTo(buffer);
+                i = tupleStr.Length;
+            }
+
+            buffer[i] = tupleSeparator;
+            i++;
 
-                var i = 0;
-                if (tupleStr != null)
+            if (Ssse3.IsSupported)
+            {
+                // These must be explicity typed as ReadOnlySpan<byte>
+                // They then become a non-allocating mappings to the data section of the assembly.
+                ReadOnlySpan<byte> shuffleMaskData = new byte[16]
```

Comment:
Is there a link we can put here explaining that this doesn't allocate?

Reply:
https://vcsjones.dev/2019/02/01/csharp-readonly-span-bytes-static/ is used in this repo for that.
E.g. https://github.com/dotnet/aspnetcore/blob/e9e6b535266f3a51828f9f272641d2fbfa98d2ff/src/Shared/runtime/Http2/Hpack/StatusCodes.cs#L12

Reply:
```suggestion
                // This uses C# compiler's ability to refer to static data directly. For more information see https://vcsjones.dev/2019/02/01/csharp-readonly-span-bytes-static 
                ReadOnlySpan<byte> shuffleMaskData = new byte[16]
```